### PR TITLE
TMA-237 Update logic to use a single loop instead of iterating over e…

### PR DESCRIPTION
**Pull request by jashelb90**
…ach logical process in order to avoid multiple loops.

Tests updated to reflect

### CODE REVIEW

File: event-processing/event-processor/app.ts [hunk 1]
```diff
@@ -1,46 +1,25 @@
...
-                            })
-                            .map((response: IValidationResponse) => {
-                                return response.error as IRequiredFieldError;
-                            }),
+        if (!validationResponse.isValid) {
+            console.log(
+                '[ERROR] VALIDATION ERROR\n' +
+                    JSON.stringify(
+                        new ValidationException(
+                            'One or more event messages failed validation.',
```
Comment (phantas) [inline event-processing/event-processor/app.ts:16]:
```suggestion
                            'An event message failed validation.',
```

File: event-processing/event-processor/tests/unit/app.test.ts [hunk 2]
```diff
@@ -363,11 +363,11 @@ describe('Unit test for app handler', function () {
...
-        expect(consoleMock).toHaveBeenNthCalledWith(4, 'Topic ARN: SOME-SNS-TOPIC');
-        expect(consoleMock).toHaveBeenNthCalledWith(5, 'MessageID is 2');
+        expect(consoleMock).toHaveBeenNthCalledWith(1, 'Topic ARN: SOME-SNS-TOPIC');
+        expect(consoleMock).toHaveBeenNthCalledWith(2, 'MessageID is 1');
+        expect(consoleMock).toHaveBeenNthCalledWith(3, 'Topic ARN: SOME-SNS-TOPIC');
+        expect(consoleMock).toHaveBeenNthCalledWith(4, 'MessageID is 2');
+        expect(consoleMock).toHaveBeenNthCalledWith(5, '[ERROR] VALIDATION ERROR\n{"requireFieldError":{"sqsResourceName":"arn:aws:sqs:us-west-2:123456789012:SQSQueue","eventId":"66258f3e-82fc-4f61-9ba0-62424e1f06b4","eventName":"","timestamp":"1609462861","requiredField":"event_name","message":"event_name is a required field."}}');
         expect(sns.publish).toHaveBeenCalledWith(
             {
                 Message: expectedResultOne,
```
Comment (phantas) [inline event-processing/event-processor/tests/unit/app.test.ts:373]: Are lines 377-381 successful and correct? I would have expected that `sns.publish` would not have been called for the invalid event, so what am I missing?

File: event-processing/event-processor/app.ts [hunk 3]
```diff
@@ -1,46 +1,25 @@
 import { SQSEvent, SQSRecord } from 'aws-lambda';
 import { ValidationService } from './services/validation-service';
 import { SnsService } from './services/sns-service';
-import { IValidationResponse } from './models/validation-response.interface';
```
Comment (phantas) [inline event-processing/event-processor/app.ts:4]: Do we still use this interface anywhere?

**Review by phantas — CHANGES_REQUESTED** [review]
A small content change and answers to some questions on a couple of tests, please.

File: event-processing/event-processor/tests/unit/app.test.ts (hunk 2 above)
Comment (jashelb90) [reply event-processing/event-processor/tests/unit/app.test.ts:373]: It includes 2 valid messages and one invalid message. So the calls to SNS publish are valid ones.

Comment (phantas) [reply event-processing/event-processor/tests/unit/app.test.ts:373]: Ah, I didn't see we were sending the same message twice. Why are we doing that? And this is a tiny nibly detail so it's fine to leave: if it's the same message, why do we have expectedResultOne and expectedResultTwo and not use just one?

**Review by phantas — DISMISSED** [review]

File: event-processing/event-processor/tests/unit/app.test.ts (hunk 2 above)
Comment (jashelb90) [reply event-processing/event-processor/tests/unit/app.test.ts:373]: That's just me missing the obvious. I will make that change to tidy it up.

Comment (jashelb90) [reply event-processing/event-processor/tests/unit/app.test.ts]: I've pushed a change that should tidy that up.

**Review by phantas — APPROVED** [review]

**Pull request merged**